adhocracy4/static/api.js

```diff
@@ -31,16 +28,13 @@ var api = (function () {
       id = null
     }
 
-    var url
-    if (urls.hasOwnProperty(endpoint)) {
-      url = urls[endpoint]
-    } else {
-      url = genericUrls[endpoint]
-        .replace('$contentTypeId', data['content_type'])
-        .replace('$objectPk', data['object_pk'])
+    var url = urls[endpoint]
+    if (data.urlReplaces) {
+      url = url.replace(/\$(\w+?)\b/g, (match, group) => {
+        return data.urlReplaces[group]
+      })
       data = $.extend({}, data)
```

Comment:
Because the caller passes you an object and we start deleting values from it. And it could be that the caller wants to do something else with the object afterwards. Than she would probably wonder what happend to those keys.